zeebe/exporters/camunda-exporter/src/main/java/io/camunda/exporter/tasks/ReschedulingTask.java

```diff
@@ -41,41 +41,43 @@ public ReschedulingTask(
 
   @Override
   public void run() {
-    var batchArchived = job.archiveNextBatch();
+    var result = task.execute();
     // while we could always expect this to return a non-null result, we don't necessarily want to
     // stop, and more importantly, we want to make it transparent that something went wrong
-    if (batchArchived == null) {
+    if (result == null) {
       logger.warn(
-          "Expected to archive a batch asynchronously, but no result returned for job {}; rescheduling anyway",
-          job);
-      batchArchived = CompletableFuture.completedFuture(0);
+          "Expected to perform a background task, but no result returned for job {}; rescheduling anyway",
+          task);
+      result = CompletableFuture.completedFuture(0);
     }
 
-    batchArchived
-        .thenApplyAsync(this::onBatchArchived, executor)
-        .exceptionallyAsync(this::onArchivingError, executor)
-        .thenAcceptAsync(this::rescheduleJob, executor);
+    result
+        .thenApplyAsync(this::onWorkPerformed, executor)
+        .exceptionallyAsync(this::onError, executor)
+        .thenAcceptAsync(this::reschedule, executor);
   }
 
-  private long onBatchArchived(final int count) {
+  private long onWorkPerformed(final int count) {
```

Comment:
❓ I dont know exactly the other jobs that we going to be used here besides the post archiver, but are we generalizing here so that in the future we can add another kind of background tasks?

Reply:
Yes, because the next kind is the `post-importer`, not `post-archiver` ;) So it doesn't archive anything, but modifies things after import.

Reply:
Ah I see, totally makes sense then!